core/src/main/java/hudson/util/io/ParserConfigurator.java

```diff
@@ -55,6 +55,7 @@
  */
 public abstract class ParserConfigurator implements ExtensionPoint, Serializable {
     private static final long serialVersionUID = -2523542286453177108L;
+    private static volatile Collection<ParserConfigurator> cache;
```

Comment:
`@Extension`s should generally not be cached in a `static` field, lest the wrong objects carry over from one `Jenkins` session to the next. If I understand correctly, this field is only used when this class is loaded in an agent JVM, which would be specific to one session, so it is not a problem.